py/desispec/joincosmics.py

```diff
@@ -15,8 +15,27 @@
 from desispec.maskbits import ccdmask
 from desispec.maskbits import specmask
 
-from skimage.transform import probabilistic_hough_line
-from skimage.morphology import binary_dilation, binary_closing
+try:
+    # Note: scikit-image is not part of desiconda.
+    from skimage.morphology import binary_closing
+except ImportError as e:
+    # If scikit-image is not available, redefine the interface.
+    def binary_dilation(image, selem=None, out=None):
+        if out is None:
+            out = np.empty(image.shape, dtype=np.bool)
+        scipy.ndimage.binary_dilation(image, structure=selem, output=out)
+        return out
+
+    def binary_erosion(image, selem=None, out=None):
+        if out is None:
+            out = np.empty(image.shape, dtype=np.bool)
+        scipy.ndimage.binary_erosion(image, structure=selem, output=out, border_value=True)
+        return out
+
+    def binary_closing(image, selem=None, out=None):
+        dilated = binary_dilation(image, selem)
+        out = binary_erosion(dilated, selem, out=out)
+        return out
 
 import matplotlib as mpl
 import matplotlib.pyplot as plt
```

Comment:
I didn't catch this until after merging, but since preprocessing now uses joincosmics, these top-level imports of matplotlib broke preprocessing at NERSC in batch mode since X11 $DISPLAY isn't available.  I fixed this with a direct push to master by moving the matplotlib imports into the functions that use them so that it is only imported if actually needed.

@sybenzvi please confirm that I didn't mess up your plotting; I only tested that I did fix the preprocessing.

Side note: Travis tests didn't catch this, but nightly integration tests at NERSC in a cronjob did.